src/Servers/Connections.Abstractions/src/Features/ITlsHandshakeFeature.cs

```diff
@@ -15,6 +19,13 @@ public interface ITlsHandshakeFeature
     /// </summary>
     SslProtocols Protocol { get; }
 
+#if NET7_0_OR_GREATER
+    /// <summary>
+    /// Gets the <see cref="TlsCipherSuite"/>.
+    /// </summary>
+    TlsCipherSuite? NegotiatedCipherSuite { get => null; }
```

Comment:
In theory we could assembly this from the other properties with a big lookup table, but the maintenance on that wouldn't be great and you'd always have some nulls for unknown combinations. I guess we can try that later.

Reply:
I considered that and concluded that it would be really, really painful.